src/watchers/indexable-post.php

```diff
@@ -0,0 +1,322 @@
+<?php
+/**
+ * WordPress Post watcher.
+ *
+ * @package Yoast\YoastSEO\Watchers
+ */
+
+namespace Yoast\YoastSEO\Watchers;
+
+use Yoast\YoastSEO\Exceptions\No_Indexable_Found;
+use Yoast\YoastSEO\WordPress\Integration;
+use Yoast\YoastSEO\Yoast_Model;
+use Yoast\YoastSEO\Models\Indexable;
+
+/**
+ * Fills the Indexable according to Post data.
+ */
+class Indexable_Post implements Integration {
+	/**
+	 * Registers all hooks to WordPress.
+	 *
+	 * @return void
+	 */
+	public function register_hooks() {
+		\add_action( 'wp_insert_post', array( $this, 'save_meta' ), PHP_INT_MAX, 1 );
+		\add_action( 'delete_post', array( $this, 'delete_meta' ) );
+	}
+
+	/**
+	 * Deletes the meta when a post is deleted.
+	 *
+	 * @param int $post_id Post ID.
+	 *
+	 * @return void
+	 */
+	public function delete_meta( $post_id ) {
+		try {
+			$indexable = $this->get_indexable( $post_id, false );
+			$indexable->delete();
+		} catch ( No_Indexable_Found $exception ) {
+			return;
+		}
+	}
+
+	/**
+	 * Saves post meta.
+	 *
+	 * @param int $post_id Post ID.
+	 *
+	 * @return void
+	 */
+	public function save_meta( $post_id ) {
+		if ( ! $this->is_post_indexable( $post_id ) ) {
+			return;
+		}
+
+		try {
+			$indexable = $this->get_indexable( $post_id );
+		} catch ( No_Indexable_Found $exception ) {
+			return;
+		}
+
+		$indexable->permalink = $this->get_permalink( $post_id );
+
+		foreach ( $this->get_meta_lookup() as $meta_key => $indexable_key ) {
+			$indexable->{$indexable_key} = $this->get_meta_value( $meta_key, $post_id );
```

Comment:
Meta values are always strings. The indexable properties are not, this will crash for example when setting `is_cornerstone` is the meta value is the string `"false"` and not a boolean.